com.playeveryware.eos/Runtime/Core/EOS_SDK_Additions/Extensions/InputStateButtonFlagsExtensions.cs

```diff
@@ -0,0 +1,131 @@
+
+    /// <summary>
+    /// Provides a means of parsing string representations of enum values for
+    /// the enum type InputStateButtonFlags.
+    /// </summary>
+    public static class InputStateButtonFlagsExtensions
+    {
+        /// <summary>
+        /// Returns the internal custom mappings.
+        /// </summary>
+        public static Dictionary<string, InputStateButtonFlags> CustomMappings { get; } = new()
+        {
+            { "DPadDown", InputStateButtonFlags.DPadDown },
+            { "DPadLeft", InputStateButtonFlags.DPadLeft },
+            { "DPadRight", InputStateButtonFlags.DPadRight },
+            { "DPadUp", InputStateButtonFlags.DPadUp },
+            { "FaceButtonBottom", InputStateButtonFlags.FaceButtonBottom },
+            { "FaceButtonLeft", InputStateButtonFlags.FaceButtonLeft },
+            { "FaceButtonRight", InputStateButtonFlags.FaceButtonRight },
+            { "FaceButtonTop", InputStateButtonFlags.FaceButtonTop },
+            { "LeftShoulder", InputStateButtonFlags.LeftShoulder },
+            { "LeftThumbstick", InputStateButtonFlags.LeftThumbstick },
+            { "LeftTrigger", InputStateButtonFlags.LeftTrigger },
+            { "RightShoulder", InputStateButtonFlags.RightShoulder },
+            { "RightThumbstick", InputStateButtonFlags.RightThumbstick },
+            { "RightTrigger", InputStateButtonFlags.RightTrigger },
+            { "SpecialLeft", InputStateButtonFlags.SpecialLeft },
+            { "SpecialRight", InputStateButtonFlags.SpecialRight }
+        };
+
+        /// <summary>
+        /// Tries to parse the given list of strings representing individual
+        /// InputStateButtonFlags enum values, and performing a
+        /// bitwise OR operation on those values.
+        /// </summary>
+        /// <param name="stringFlags">
+        /// List of strings representing individual InputStateButtonFlags enum values.
+        /// </param>
+        /// <param name="result">
+        /// The result of performing a bitwise OR operation on the values that
+        /// are represented by the list of string values.
+        /// </param>
+        /// <returns>
+        /// True if all the list of strings was successfully parsed, and the
+        /// resulting list of enum values was bitwise ORed together.
+        /// </returns>
+        public static bool TryParse(IList<string> stringFlags, out InputStateButtonFlags result)
+        {
+            return EnumUtility<InputStateButtonFlags>.TryParse(stringFlags, CustomMappings, out result);
+        }
+
+        /// <summary>
+        /// Parses an InputStateButtonFlags value in to its component strings.
+        /// These strings can then be safely stored in a configuration value.
+        /// </summary>
+        /// <param name="flags">Enum to separate into parts.</param>
+        /// <returns>A list of strings representing these flags.</returns>
+        public static List<string> FlagsToStrings(this InputStateButtonFlags flags)
```

Comment:
Per my other comment regarding the place where this function is called - this function can be removed.